src/BrickbattleWeapons/MainModule/ServerModules/slices/BrickbattleWeaponSlice.lua

```diff
@@ -0,0 +1,42 @@
+--!strict
+local Reflex = require("Packages/reflex")
+local Immut = require("Packages/immut")
+local Table = require("src/BrickbattleWeapons/MainModule/ServerModules/slices/BrickbattleWeaponTables")
+
+export type BrickbattleWeaponStateActions = {
+    setStatState: (playerName: string, numb: number, stat: string) -> (),
+    removeStatState: (playerName: string, stat: string) -> (),
+}
+
+type Data = {
+    [string]: Table.BrickbattlePlayerTable
+}
+local initialState: Data = {}
+local BrickbattlePlayerStateSlice = Reflex.createProducer(initialState, {
+    setPlayerWeaponState = function(state, playerName: string,weapon: string, Data: any)
+		return Immut.produce(state, function(draft) 
+            local playerData = draft[playerName]
```

Comment:
indentation here is weird. try to make this look better -- maybe separate out the call to Immut in a new helper function.